vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/AndroidDeviceTestRunner.kt

```diff
@@ -68,14 +70,25 @@ class AndroidDeviceTestRunner(private val device: AdamAndroidDevice) {
                             return@withTimeoutOrNull
                         } else {
                             processEvents(update, listener)
+                            when (configuration.strictMode) {
```

Comment:
@Malinskiy We've discussed that the better way is having this logic in a listener but I can't `return` from listener

Reply:
You should also close the request here

Reply:
As I understand we have `finally` block in that `channel` will be closed when we `return`

Reply:
You are right :+1: